### PR TITLE
Upgrade codecov V1 => V3 and switch to pcov

### PR DESCRIPTION
**Pull request by Chris53897**
… Climate Test Reporter

fix ci deprecation for codecov V1 => V3
https://github.com/marketplace/actions/codecov

The change from xdebug to pcov (and only one coverage output file) speed the coverage run up from > 8:32 minutes to  1:15 minutes
Example of xdebug: https://github.com/briannesbitt/Carbon/actions/runs/6021629511/job/16336221316

Example of pcov: https://github.com/Chris53897/Carbon/actions/runs/6024021872/job/16341911483

But needs testing if CodeCov works correct and Code Climate starts to work.
@kylekatarnls This can only be done from maintainers. As i got the warning/error for the missing secret ;)

Code climate does not work in the past, if we can trust the warning, that it could not find the clover.xml (because it was coverage.xml before). And it it not possible to set the filename in code climate acction.

```
Coverage - PHP 8.2 - stable - ubuntu
Command failed: cc-test-reporter after-build -t clover
time="2023-08-26T21:07:20Z" level=error msg="could not find coverage file \ncould not find any files in search paths for clover. search paths were: , build/logs/clover.xml, clover.xml" 
Error: could not find any files in search paths for clover. search paths were: , build/logs/clover.xml, clover.xml```

### CODE REVIEW

**Comment by kylekatarnls** [comment]
Thanks.

The uncovered part are just class loaders, it would be fine not to have them explicitly marked as covered. Can you add `// @codeCoverageIgnoreStart` and `// @codeCoverageIgnoreEnd` to wrap those blocks of code?

https://app.codecov.io/gh/briannesbitt/Carbon/pull/2842/blob/src/Carbon/Traits/Localization.php
https://app.codecov.io/gh/briannesbitt/Carbon/pull/2842/blob/src/Carbon/MessageFormatter/MessageFormatterMapper.php

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -153,7 +153,7 @@ jobs:
       - name: Run test suite
         run: |
           if [[ "${{ matrix.laravel }}" != 'true' && "${{ matrix.coverage }}" = 'true' ]]; then
-            php -d memory_limit=-1 -d zend.enable_gc=0 -d error_reporting=-1 vendor/phpunit/phpunit/phpunit --coverage-clover=coverage.xml --coverage-text;
+            php -d memory_limit=-1 -d zend.enable_gc=0 -d error_reporting=-1 vendor/phpunit/phpunit/phpunit --coverage-clover=clover.xml;
```
Comment (kylekatarnls) [inline .github/workflows/tests.yml:156]:
```suggestion
            php -d memory_limit=-1 -d zend.enable_gc=0 -d error_reporting=-1 vendor/phpunit/phpunit/phpunit --coverage-clover=clover.xml --coverage-text;
```

Comment (Chris53897) [reply .github/workflows/tests.yml:156]: of course, i can change this, and see how the impact on runtime is.
Can you please explain why this is necessary?

Comment (Chris53897) [reply .github/workflows/tests.yml]: There is just a little overhead for  `--coverage-text`
Time: 01:35.635, Memory: 1.04 GB

Forget the last question. You want to see the coverage report directly in the github action run.

**Review by kylekatarnls — APPROVED** [review]

**Pull request merged**